### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: src/kata1.fizzBuzz.js [hunk 1]
```diff
@@ -1,3 +1,23 @@
-const fizzBuzz = (number) => {};
+const fizzBuzz = (number) => {
```
Comment (StuartCowley) [inline src/kata1.fizzBuzz.js:1]: The code itself here is all fine, though you really dont need to add all the comments here as its already clear what you are doing. Comments should really be used to explain _why_ something is being done the way it is and not _how_, mainly when you are doing something that is not obvious at first sight or something like a complex regular expression which is not easily readable at first sight. In this case all these could be removed

File: src/kata2.booleanToWord.js [hunk 2]
```diff
@@ -1,3 +1,14 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+  if (boolean === true) {
+    return "Yes";
+  }
+  if (boolean === false) {
+    return "No";
+  }
+  return "not a boolean";
```
Comment (StuartCowley) [inline src/kata2.booleanToWord.js:8]: Nice to see you thinking about cases other than passing in boolean true or false! Edge cases are always worth thinking about as it makes your code more robust and less prone to bugs, but for this simple kata we can assume it will always be a proper boolean. In that case, this can be replaced simply with a ternary operator:
```
  return boolean === true ? "Yes" : "No";
```

File: src/kata3.numberToReversedDigits.js [hunk 3]
```diff
@@ -1,3 +1,15 @@
-const numberToReversedDigits = (number) => {};
+const numberToReversedDigits = (number) => {
+  return String(number).split("").reverse().map(Number);
+
```
Comment (StuartCowley) [inline src/kata3.numberToReversedDigits.js:3]: You dont need to include blank lines at the end of a function, this should be taken out

File: src/kata4.humanCatDogYears.js [hunk 4]
```diff
@@ -1,3 +1,22 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:1]: While this works, the logic isnt that clear. Since there are only 3 changes (for 0, 1, and 2 or more years old) you should only need 3 if statements:
```
  if (number === 0) {
    return [number, number, number];
  }
  if (number === 1) {
    return [number, number + 14, number + 14];
  }
  if (number > 1) {
    return [number, number * 4 + 16, number * 5 + 14];
  }
```

File: src/kata6.joinNames.js [hunk 5]
```diff
@@ -1,3 +1,18 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+   return namesObj.map(object => object.name).join(", ").replace(/,(?=[^,]+$)/, " &");
+
```
Comment (StuartCowley) [inline src/kata6.joinNames.js:3]: Dont need this blank line here

File: src/kata6.joinNames.js [hunk 6]
```diff
@@ -1,3 +1,18 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+   return namesObj.map(object => object.name).join(", ").replace(/,(?=[^,]+$)/, " &");
```
Comment (StuartCowley) [inline src/kata6.joinNames.js:2]: Nice regex! Do you understand what it is doing? This would be a good example of when to use a comment, as at first glance it is not clear what is happening here

File: src/kata6.joinNames.js [hunk 7]
```diff
@@ -1,3 +1,18 @@
...
+
+};
+
```
Comment (StuartCowley) [inline src/kata6.joinNames.js:5]: Dont need both of these blank lines here, one can be removed

File: src/kata7.getEmployerRole.js [hunk 8]
```diff
@@ -1,3 +1,19 @@
-const getEmployerRole = (employeeName, employees) => {};
+const getEmployerRole = (employeeName, employees) => {
+  const employee = employees.find((object) => object.name === employeeName);
 
+  return employee.role;
```
Comment (StuartCowley) [inline src/kata7.getEmployerRole.js:4]: Can simplify this a touch:
```
  return employees.find((object) => object.name === employeeName).role;
```

File: test/kata1.fizzBuzz.test.js [hunk 9]
```diff
@@ -1,11 +1,23 @@
 const { fizzBuzz } = require("../src");
 
 describe("fizzBuzz", () => {
-  test("returns Fizz when passed a multiple of 3", () => {});
+  it("returns Fizz when passed a multiple of 3", () => {
+    expect(fizzBuzz(3)).toBe("Fizz");
+  })
```
Comment (StuartCowley) [inline test/kata1.fizzBuzz.test.js:6]: Just one blank line in between your tests is fine

File: test/kata2.booleanToWord.test.js [hunk 10]
```diff
@@ -1,5 +1,17 @@
...
-  // how do we create specs again???
+  it("returns the word Yes when passed a true value", () => {
+    expect(booleanToWord(true)).toBe("Yes");
+  });
+
+  it("returns the word No when passed a false value", () => {
+    expect(booleanToWord(false)).toBe("No");
+  });
+
+  it("returns not a boolean when passed a non boolean value", () => {
```
Comment (StuartCowley) [inline test/kata2.booleanToWord.test.js:12]: Nice edge case testing :)

File: test/kata3.numberToReversedDigits.test.js [hunk 11]
```diff
@@ -1,5 +1,8 @@
 const { numberToReversedDigits } = require("../src");
 
 describe("numberToReversedDigits", () => {
-  test("returns a reversed array of the number's digits", () => {});
-});
\ No newline at end of file
+  it("returns a reversed array of the number's digits", () => {
+    expect(numberToReversedDigits(12345)).toEqual([5, 4, 3, 2, 1]);
+    expect(numberToReversedDigits(582)).toEqual([2,8,5])
```
Comment (StuartCowley) [inline test/kata3.numberToReversedDigits.test.js:6]: Could add a test here for edge cases, like a single digit number being passed in, or if feeling more adventurous how would you would a negative number (if at all)?

File: test/kata6.joinNames.test.js [hunk 12]
```diff
@@ -1,5 +1,17 @@
 const { joinNames } = require("../src");
 
 describe("joinNames", () => {
-  test("returns string of names, seperated by commas and an ampersand", () => {});
+  it("returns string of names, seperated by commas and an ampersand", () => {
```
Comment (StuartCowley) [inline test/kata6.joinNames.test.js:4]: Would be good practice to add another test or 2 with different length arrays being passed in here

File: test/kata7.getEmployerRole.test.js [hunk 13]
```diff
@@ -1,5 +1,27 @@
...
+  
+      role: "Sales Associate",
+  
+  }, {
+        name: "Javid",
+
+  
+      role: "Human Recommended Reading Assistant",
+  
+  }];
```
Comment (StuartCowley) [inline test/kata7.getEmployerRole.test.js:22]: Formatting is a bit out here, theres a lot of extra blank lines and the indentation is a bit scatty. Should look like this:
```
    const employees = [
      {
        name: "Satti",
        role: "Developer",
      },
      {
        name: "Jenny",
        role: "Sales Associate",
      },
      {
        name: "Javid",
        role: "Human Recommended Reading Assistant",
      },
    ];
```

File: test/kata7.getEmployerRole.test.js [hunk 14]
```diff
@@ -1,5 +1,27 @@
...
+    expect(getEmployerRole("Javid", employees)).toEqual(
```
Comment (StuartCowley) [inline test/kata7.getEmployerRole.test.js:23]: Always good to have more than one assertion here just to make sure!

**Review by StuartCowley — COMMENTED** [review]
Good stuff here, a lot of detail too though it really isnt necessary. You have almost as much space taken up with comments as you do with code! You could remove basically all the comments here as you are working with standard JS methods and functions so they dont require an explanation (with one exception being the regex, have left a comment about that specifically)
It would also be good to make sure you run your code through a linter before submitting as the formatting is a bit out in places, nothing severe but definitely worth mentioning as this is something that will always be important and necessary in the job. 
All the tests pass, so just a few tweaks and you'll have a much better repo :)
